packages/ui/lit-grid/src/dx-grid.ts

```diff
@@ -1228,19 +1228,52 @@ export class DxGrid extends LitElement {
 
     const cellReadonly = this.cell(col, row, plane)?.readonly;
     if (cellReadonly !== undefined) {
-      return cellReadonly;
+      // Convert the possible string values to boolean
+      return cellReadonly === 'no-text-select' || cellReadonly === 'text-select' || cellReadonly === true;
     }
 
-    return (
-      (this.columns?.[colPlane]?.[col]?.readonly ?? this.columnDefault?.[colPlane]?.readonly) ||
-      (this.rows?.[rowPlane]?.[row]?.readonly ?? this.rowDefault?.[rowPlane]?.readonly)
-    );
+    const colDefault = this.columns?.[colPlane]?.[col]?.readonly ?? this.columnDefault?.[colPlane]?.readonly;
+    const rowDefault = this.rows?.[rowPlane]?.[row]?.readonly ?? this.rowDefault?.[rowPlane]?.readonly;
+
+    // Ensure we return a boolean by checking if either value is truthy
+    return !!colDefault || !!rowDefault;
+  }
+
+  /**
+   * Determines if the cell's text content should be selectable based on its readonly value.
+   * @returns true if the cells text content is selectable, false otherwise.
+   */
+  private cellTextSelectable(col: number, row: number, plane: DxGridPlane): boolean {
+    const colPlane = resolveColPlane(plane);
+    const rowPlane = resolveRowPlane(plane);
+
+    // Check cell-specific setting first.
+    const cellReadonly = this.cell(col, row, plane)?.readonly;
+    if (cellReadonly !== undefined) {
+      return cellReadonly === 'text-select' || cellReadonly === false;
+    }
+
+    // Check column/row defaults.
+    const colDefault = this.columns?.[colPlane]?.[col]?.readonly ?? this.columnDefault?.[colPlane]?.readonly;
+    const rowDefault = this.rows?.[rowPlane]?.[row]?.readonly ?? this.rowDefault?.[rowPlane]?.readonly;
+    if (colDefault === 'text-select' || rowDefault === 'text-select') {
+      return true;
+    }
+    if (colDefault === true || colDefault === 'no-text-select') {
+      return false;
+    }
+    if (rowDefault === true || rowDefault === 'no-text-select') {
+      return false;
+    }
+
+    return !this.cellReadonly(col, row, plane);
```

Comment:
text is only selectable if the `readonly` value is exactly `text-select`, if the cell is not readonly then it’s up to the consumer to handle displaying an editor since dx-grid doesn’t support `contenteditable` on its own.

Reply:
100%. Fixed this. Thanks!